docs/documentation/api/Server.md

```diff
@@ -24,6 +24,10 @@ A config object with the following options:
 
 3. `transport` (_object_): the transport implementation.
    If not provided, socket.io is used.
+   
+4. `generateCredentials` (_function_): an optional function that returns player credentials to store in the game metadata and validate against. If not specified, the Lobby’s `uuid` implementation will be used.
```

Comment:
I feel like there is opportunity to merge `generateCredentials` and `uuid` into one thing.  Is there value in them being separate?

Reply:
`uuid` is currently used to generate both player credentials and game IDs, which seem like quite separate things to me. That was fine for a context-less random generator, but I’m passing Koa’s `ctx` to `generateCredentials`, so that changes things a bit.

I also chose to have `generateCredentials` fall back first to `lobbyConfig.uuid` and then to `shortid.generate` as that preserves current behaviour, but falling back to `shortid` directly would probably make more sense. (The `lobbyConfig.uuid` option was misdocumented as `shortid` before, so I’m guessing there probably wouldn’t be too many people impacted if that changed.)

Reply:
Sorry if I missed some context in the original issue discussing this, but what's the use-case for having a customized function to generate credentials that isn't just invoking something like `uuid`?

Reply:
The example I gave in the docs is of calling into an authentication service.

For example, Firebase Auth allows you to get an ID token for a logged in user on the client. You could then send that in the join game request, have `generateCredentials` decode the token, and return the decoded user ID (which is public) to be stored in the metadata. Then when making a move, the client could send along a similar (potentially refreshed) token and the `authenticateCredentials` method could again decode that token, and check if it matches the user ID stored in the metadata.

That could provide a way for users to authenticate that doesn’t rely on session storage.

Reply:
Makes sense.  Thanks for clarifying!